crates/rvm-rs/src/releases.rs

```diff
@@ -0,0 +1,298 @@
+                                pre: Prerelease::default(),
+                            },
+                        ],
+                    }
+                    .to_string(),
+                )
+                .finish(),
+            Binary::Remote(info) => f
+                .debug_struct("Remote")
+                .field("version", &info.version.to_string())
+                .field(
+                    "solc_req",
+                    &semver::VersionReq {
+                        comparators: vec![
+                            Comparator {
+                                op: semver::Op::GreaterEq,
+                                major: info.first_supported_solc_version.major,
+                                minor: Some(info.first_supported_solc_version.minor),
+                                patch: Some(info.first_supported_solc_version.patch),
+                                pre: Prerelease::default(),
+                            },
+                            Comparator {
+                                op: semver::Op::LessEq,
+                                major: info.last_supported_solc_version.major,
+                                minor: Some(info.last_supported_solc_version.minor),
+                                patch: Some(info.last_supported_solc_version.patch),
+                                pre: Prerelease::default(),
+                            },
+                        ],
+                    }
+                    .to_string(),
+                )
+                .finish(),
+        }
+    }
+}
+
+impl Binary {
+    /// Returns the version for the given `Binary`
+    pub fn version(&self) -> &Version {
+        match self {
+            Binary::Local { info, .. } => &info.version,
+            Binary::Remote(info) => &info.version,
+        }
+    }
+    /// Returns the path for the given `Binary`
+    pub fn local(&self) -> Option<&Path> {
+        match self {
+            Binary::Local { path, .. } => Some(path.as_ref()),
+            Binary::Remote(_) => None,
+        }
+    }
+}
+
+/// Basic information about Resolc build that is available to be installed
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Build {
+    pub(crate) name: String,
+    pub(crate) version: Version,
+    #[serde(rename = "longVersion")]
+    pub(crate) long_version: String,
+    pub(crate) url: Url,
+    #[serde(rename = "firstSolcVersion")]
+    pub(crate) first_supported_solc_version: Version,
+    #[serde(rename = "lastSolcVersion")]
+    pub(crate) last_supported_solc_version: Version,
+}
+
+impl Build {
+    /// Checks compatibility between selected Resolc and `solc` versions
+    ///
+    /// # Arguments
+    ///
+    /// * `solc_version` -  `solc` version requirement this will allow check the compatibility between the two compiler versions
+    pub fn check_solc_compat(&self, solc_version: &Version) -> Result<(), Error> {
```

Comment:
I assume that the logic for downloading the supported solc compiler will be in foundry.
Just thinking if it make sense to use the svm-rs from rvm-rs to make the full revive setup in one place.

Reply:
solc is a dependency of resolc. I'd expect any version manager to also handle the dependencies of the things they manage.

Reply:
@pkhry I think it would be very useful especially for binary rvm-rs version. but we can add this support later as separate PR.